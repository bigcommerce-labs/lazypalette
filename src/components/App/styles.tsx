import styled from 'styled-components';

export const StyledApp = styled.div`
    background: ${({ theme: { colors } }) => colors.background};
    align-content: flex-start;
    display: flex;
    flex-direction: column;
    height: 100%;
    position: fixed;
    width: 100%;
`;

export const Viewport = styled.div`
    display: flex;
    flex-direction: row;
    flex-grow: 1;
`;
