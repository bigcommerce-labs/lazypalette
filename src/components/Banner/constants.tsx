import React from 'react';

export const CopyMessage = 'Link copied to clipboard.';

export const TooltipMessage =
    <>
        <p>You must launch your store from the Getting Started page in order to make it public.</p>
        <p><a href="/manage/getting-started">Go to Getting Started</a></p>
        <p>You can still let other people see your store before it is launched by sharing the private store link.</p>
    </>;

export const MaintenanceMessage = 'Your store is down for maintenance.';
export const PrelaunchMessage = 'Your store has not launched and is not publicly visible.';
