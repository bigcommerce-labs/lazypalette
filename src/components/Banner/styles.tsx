import styled from 'styled-components';

export const StyledBanner = styled.div`
    background-color: #FFD24D;
    text-align: center;
    width: 100%;
    line-height: 2.5rem;
    font-size: 1rem;
    font-weight: 600;
    color: #303540;
`;

export const StyledIcon = styled.span`
    span {
        margin-bottom: -5px;
    }
`;

export const StyledMessage = styled.p`
    text-align: left;

    a {
        color: ${({ theme }) => theme.colors.primary};
        display: block;
        font-weight: bold;
        margin-top: 5px;
        text-decoration: none;
    }
`;

export const StyledShareableLink = styled.button`
    background: ${({ theme }) => theme.colors.warning};
    color: ${({ theme }) => theme.colors.primaryText};
    font-size: ${({ theme }) => theme.typography.fontSize.small};
    font-weight: ${({ theme }) => theme.typography.fontWeight.bold}
    font-family: ${({ theme }) => theme.typography.fontFamily.headings};
    margin: .5rem 0 .5rem 1.25rem;
    outline: none;
    border: 1px solid;
    border-radius: 2px;
    display: inline-block;
    padding: .25rem .5rem;
    line-height: 1.375rem;
    cursor: pointer;
`;
