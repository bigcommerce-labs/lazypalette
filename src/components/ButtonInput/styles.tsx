import styled from 'styled-components';

export const Wrapper = styled.div`
    margin: 0;
`;

interface ButtonProps {
    border?: boolean;
    classType?: string;
}

export const Button = styled.button.attrs<ButtonProps>({})`
    height: 2rem;
    border-radius: 2px;
    box-sizing: border-box;
    border: ${({ border, theme }) => border ? `1px solid ${theme.colors.primary}` : 'none'};
    padding: ${({ border }) => border ? '0 1rem' : '0'};
    font-family: ${({ theme }) => theme.typography.fontFamily.sans };
    font-size: 1rem;
    cursor: pointer;

    ${({ border, classType, theme }) => {
        const isPrimary = classType === 'primary';

        return `
            background-color: ${isPrimary ? theme.colors.primary : '#FFF'};
            color: ${isPrimary ? '#FFF' : theme.colors.primary};

            :hover {
                background-color: ${isPrimary ? theme.colors.primaryHover : theme.colors.secondaryHover};
                ${!border && `
                    background-color: transparent;
                `}
                border-color: ${isPrimary ? theme.colors.brandPrimary : theme.colors.primary};
                text-decoration: ${border ? 'none' : 'underline'};
            }

            :focus {
                outline: none !important;
            }

            :disabled {
                background-color: ${isPrimary ? theme.colors.primary : '#FFF'};
                opacity: 0.5;
            }
        `;
    }}
`;

Button.defaultProps = {
    theme: {
        colors: {
            background: '#F5F7FA',
            brandPrimary: '#273A8A',
            primary: '#3F69FE',
        },
        typography: {
            fontFamily: {
                sans: '"Source Sans Pro", "Helvetica Neue", Arial, sans-serif',
            },
        },
    },
};
