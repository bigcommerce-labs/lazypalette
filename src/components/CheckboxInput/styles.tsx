import styled from 'styled-components';

export const Container = styled.div`
    position: relative;
`;

export const Input = styled.input.attrs({
    type: 'checkbox',
})`
    ${({ theme }) => `
        opacity: 0;

        :not(:checked) + label:after {
            opacity: 0;
        }

        :not(:checked) + label {
            background: #FFF;
            border: 1px solid ${theme.colors.guideText};
            box-sizing: border-box;
        }
    `}
`;

export const Label = styled.label`
    ${({ theme }) => `
        cursor: pointer;
        position: absolute;
        width: 1.25rem;
        height: 1.25rem;
        top: 25%;
        left: 0;
        background: ${theme.colors.primary};
        border-radius: 2px;

        :after {
            content: '';
            position: absolute;
            width: .5rem;
            height: .25rem;
            background: transparent;
            top: .25rem;
            left: .25rem;
            border: 3px solid ${theme.colors.stroke};
            border-top: none;
            border-right: none;
            transform: rotate(-45deg);
        }
    `}
`;

export const HiddenLabel = styled.label`
    padding-left: .5rem;
    line-height: 1.25rem;
    display: inline-block;
`;
