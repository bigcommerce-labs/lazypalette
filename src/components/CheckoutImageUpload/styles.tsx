import styled from 'styled-components';

export const UploadButtonLabel = styled.label`
    height: 2rem;
    border-radius: 2px;
    border: 1px solid ${({ theme }) => theme.colors.primary};
    padding: 0 0.75em;
    font-size: 0.75rem;
    cursor: pointer;
    display: inline-block;
    color: ${({ theme }) => theme.colors.primary};
    margin-right: 1rem;
`;

export const CheckoutImageUploadWrapper = styled.div`
    margin: 1rem 0;
`;

export const CheckoutImageInputWrapper = styled.div`
    margin: 1rem 0;
`;

export const CheckoutImagePreviewWrapper = styled.div``;

export const CheckoutImageUploadError = styled.p`
    color: ${({ theme }) => theme.colors.error};
`;

export const CheckoutImageRemoveLink = styled.a`
    color: ${({ theme }) => theme.colors.error};
`;
