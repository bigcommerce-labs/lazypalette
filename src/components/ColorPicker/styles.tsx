import styled from 'styled-components';

export const Container = styled.div`
  margin: 0;
`;

interface SelectedColorProps {
  color: any;
}

export const SelectedColor = styled.div.attrs<SelectedColorProps>({}) `
  background: ${(props: SelectedColorProps) => `rgba(
    ${props.color.r},
    ${props.color.g},
    ${props.color.b},
    ${props.color.a}
  )`};
  border-radius: 3px;
  height: 1.5rem;
  width: 1.5rem;
  cursor: pointer;
  margin: 1px;
`;
