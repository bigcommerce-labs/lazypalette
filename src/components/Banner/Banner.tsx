import clipboardCopy from 'clipboard-copy';
import { theme, Icon } from 'pattern-lab';
import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';

import { HeaderMenuMessages } from '../HeaderMenu/constants';
import ConfirmModal from '../Modal/ConfirmModal/ConfirmModal';
import Tooltip from '../Tooltip/Tooltip';

import { ToastType } from '../../actions/constants';
import { State } from '../../reducers/reducers';
import { trackCopyPrivateLink } from '../../services/analytics';

import {
    BannerLinks,
    CopyButtonText,
    CopyMessage,
    MaintenanceMessage,
    PrelaunchMessage,
    TooltipEnd,
    TooltipLead,
    TooltipLink,
} from './constants';
import { StyledBanner, StyledIcon, StyledLink, StyledMessage, StyledShareableLink } from './styles';

interface BannerProps {
    isChanged: boolean;
    isDownForMaintenance: boolean;
    isPrelaunchStore: boolean;
    previewCode: string;
    shopPath: string;
    createNotification(autoDismiss: boolean, message: string, type: string): Dispatch<State>;
}

interface BannerState {
    isUnsavedOpen: boolean;
}

export class Banner extends PureComponent<BannerProps> {
    readonly state: BannerState = {
        isUnsavedOpen: false,
    };

    handleLinkClick = () => {
        const { isChanged } = this.props;

        if (isChanged) {
            this.setState({ isUnsavedOpen: true });
        } else {
            this.handleModalLink();
        }
    };

    handleShareableClick = () => {
        const { createNotification, previewCode, shopPath } = this.props;

        trackCopyPrivateLink(CopyButtonText);
        clipboardCopy(`${shopPath}/?guestTkn=${previewCode}`);
        createNotification(true, CopyMessage, ToastType.Success);
    };

    handleModalCancel = () => this.setState({ isUnsavedOpen: false });

    handleModalLink = () => window.location.assign(BannerLinks.GettingStarted);

    render() {
        const { isDownForMaintenance, isPrelaunchStore } = this.props;

        if (!isDownForMaintenance && !isPrelaunchStore) {
            return null;
        }

        const { isUnsavedOpen } = this.state;
        const message = isDownForMaintenance ? MaintenanceMessage : PrelaunchMessage;

        return (
            <StyledBanner>
                {message} {(!isDownForMaintenance && isPrelaunchStore) &&
                    <Tooltip
                        clickable
                        message={
                            <StyledMessage>
                                <p>{TooltipLead}</p>
                                <StyledLink onClick={this.handleLinkClick}>
                                    {TooltipLink}
                                </StyledLink>
                                <p>{TooltipEnd}</p>
                            </StyledMessage>
                        }
                    >
                        <StyledIcon>
                            <Icon
                                glyph={'help'}
                                primaryColor={theme.colors.primaryText}
                                size={'large'}
                            />
                        </StyledIcon>
                    </Tooltip>
                }
                {!isDownForMaintenance &&
                    <StyledShareableLink onClick={this.handleShareableClick}>
                        {CopyButtonText}
                    </StyledShareableLink>
                }
                {isUnsavedOpen &&
                    <ConfirmModal
                        primaryAction={this.handleModalLink}
                        primaryActionText={HeaderMenuMessages.ModalAction}
                        secondaryAction={this.handleModalCancel}
                        overlayClose={this.handleModalCancel}
                        title={HeaderMenuMessages.ModalTitle}
                    >
                        {HeaderMenuMessages.UnsavedModalBody}
                    </ConfirmModal>
                }
            </StyledBanner>
        );
    }
}

const mapStateToProps = (state: State) => ({
    isChanged: state.theme.isChanged,

});

export default connect(mapStateToProps)(Banner);
