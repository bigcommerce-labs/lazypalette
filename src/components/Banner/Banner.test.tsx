import { mount, shallow } from 'enzyme';
import React from 'react';
import { ThemeProvider } from 'styled-components';

import ConfirmModal from '../Modal/ConfirmModal/ConfirmModal';
import Tooltip from '../Tooltip/Tooltip';

import { BannerLinks } from './constants';
import { StyledLink } from './styles';
import { Banner } from './Banner';

describe('Banner', () => {
    const theme = {
        colors: {},
        elevation: {},
        layers: {},
        typography: {
            fontFamily: {},
            fontSize: {},
            fontWeight: {},
        },
    };

    describe('when rendering', () => {
        it('renders correctly in live mode', () => {
            const mockNotify = jest.fn();

            const buttonInput = shallow(
                <Banner
                    createNotification={mockNotify}
                    previewCode="abcdefg"
                    isChanged={false}
                    isPrelaunchStore={false}
                    isDownForMaintenance={false}
                    shopPath="http://catland.bigcommerce.com"
                />
            );

            expect(buttonInput).toMatchSnapshot();
        });

        it('renders correctly in prelaunch mode', () => {
            const mockNotify = jest.fn();

            const buttonInput = shallow(
                <Banner
                    createNotification={mockNotify}
                    previewCode="abcdefg"
                    isChanged={false}
                    isPrelaunchStore={true}
                    isDownForMaintenance={false}
                    shopPath="http://catland.bigcommerce.com"
                />
            );

            expect(buttonInput).toMatchSnapshot();
        });

        it('renders correctly in maintenance mode', () => {
            const mockNotify = jest.fn();

            const buttonInput = shallow(
                <Banner
                    createNotification={mockNotify}
                    previewCode="abcdefg"
                    isChanged={false}
                    isPrelaunchStore={false}
                    isDownForMaintenance={true}
                    shopPath="http://catland.bigcommerce.com"
                />
            );

            expect(buttonInput).toMatchSnapshot();
        });
    });

    describe('when tooltip is clicked', () => {
        describe('when message link is clicked and isChanged is true', () => {
            it('should open up the confirm dialog modal', () => {
                const mockNotify = jest.fn();

                const buttonInput = mount(
                    <ThemeProvider theme={theme}>
                        <Banner
                            createNotification={mockNotify}
                            previewCode="abcdefg"
                            isChanged={true}
                            isPrelaunchStore={true}
                            isDownForMaintenance={false}
                            shopPath="http://catland.bigcommerce.com"
                        />
                    </ThemeProvider>
                );

                expect(buttonInput.find(ConfirmModal).exists()).toEqual(false);
                buttonInput.find(Tooltip).simulate('click'); // open tooltip
                buttonInput.find(StyledLink).simulate('click'); // click message link
                expect(buttonInput.find(ConfirmModal).exists()).toEqual(true);
            });
        });

        describe('when message link is clicked and isChanged is false', () => {
            it('should invoke window.location.assign, and not confirm modal', () => {
                const mockNotify = jest.fn();

                const buttonInput = mount(
                    <ThemeProvider theme={theme}>
                        <Banner
                            createNotification={mockNotify}
                            previewCode="abcdefg"
                            isChanged={false}
                            isPrelaunchStore={true}
                            isDownForMaintenance={false}
                            shopPath="http://catland.bigcommerce.com"
                        />
                    </ThemeProvider>
                );

                const _window: any = window;
                _window.location.assign = jest.fn();

                expect(_window.location.assign).not.toBeCalled();
                buttonInput.find(Tooltip).simulate('click'); // open tooltip
                buttonInput.find(StyledLink).simulate('click'); // click message link

                expect(buttonInput.find(ConfirmModal).exists()).toEqual(false);
                expect(_window.location.assign).toBeCalledWith(
                    BannerLinks.GettingStarted
                );
            });
        });
    });
});
