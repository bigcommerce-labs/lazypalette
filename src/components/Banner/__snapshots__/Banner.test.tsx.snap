// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner when rendering renders correctly in live mode 1`] = `""`;

exports[`Banner when rendering renders correctly in maintenance mode 1`] = `
<styled.div>
  Your store is down for maintenance.
   
</styled.div>
`;

exports[`Banner when rendering renders correctly in prelaunch mode 1`] = `
<styled.div>
  Your store has not launched and is not publicly visible.
   
  <Tooltip
    clickable={true}
    message={
      <styled.div>
        <p>
          You must launch your store from the Getting Started page in order to make it public.
        </p>
        <styled.a
          onClick={[Function]}
        >
          Go to Getting Started
        </styled.a>
        <p>
          
    You can still let other people see your store before it is launched by sharing the private store link.

        </p>
      </styled.div>
    }
  >
    <styled.span>
      <Icon
        glyph="help"
        primaryColor="#303540"
        size="large"
      />
    </styled.span>
  </Tooltip>
  <styled.button
    onClick={[Function]}
  >
    Copy private store link
  </styled.button>
</styled.div>
`;
